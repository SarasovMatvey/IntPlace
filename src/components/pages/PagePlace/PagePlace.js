import { useEffect } from "react";
import { useState } from "react";
import { useParams } from "react-router-dom";
import { Container } from "semantic-ui-react";
import sygicAxios from "../../../api/sygic";
import PlaceInfo from "../../organisms/PlaceInfo";
import BaseTemplate from "../../templates/BaseTemplate/BaseTemplate";
import "./PagePlace.sass";

function PagePlace() {
  const { id: placeId } = useParams();
  const [placeInfo, setPlaceInfo] = useState(null);

  useEffect(() => {
    sygicAxios(`/places/${placeId}`).then(({ data: { data } }) => {
      setPlaceInfo(data.place);
    });
  }, [placeId]);

  return (
    <BaseTemplate>
      <div className="page-place">
        <Container>
          {placeInfo ? (
            <PlaceInfo
              placeName={placeInfo.name}
              mainImg={placeInfo.main_media?.media[0].url}
              description={placeInfo.description?.text}
              media={placeInfo.main_media?.media
                .slice(1)
                .map(mediaData => mediaData.url)}
              params={{
                Address: placeInfo.address,
                Rating: placeInfo.rating.toFixed(1),
                Phone: placeInfo.phone,
                Email: placeInfo.email,
              }}
            />
          ) : null}
        </Container>
      </div>
    </BaseTemplate>
  );
}

export default PagePlace;
