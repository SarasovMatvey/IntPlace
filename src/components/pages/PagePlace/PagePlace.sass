.page-place
  &__place-info 
    margin-bottom: 3rem
  &__near-places
    padding-left: 2rem