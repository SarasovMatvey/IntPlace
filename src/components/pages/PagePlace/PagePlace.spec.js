import PagePlace from "./PagePlace";

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useParams: () => ({
    id: "poi:201",
  }),
}));

const setUp = props => shallow(<PagePlace {...props} />);

describe("PagePlace", () => {
  let component;

  beforeEach(() => {
    component = setUp();
  });

  it("should renders properly", () => {
    expect(component).toMatchSnapshot();
  });
});
