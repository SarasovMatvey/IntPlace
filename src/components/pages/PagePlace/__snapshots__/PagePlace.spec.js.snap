// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PagePlace should renders properly 1`] = `
<BaseTemplate>
  <div
    className="page-place"
  >
    <Container />
  </div>
</BaseTemplate>
`;
